flink-table/flink-table-planner-blink/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/serde/LogicalTypeSerdeTest.java

```diff
@@ -105,7 +105,7 @@ public void testLogicalTypeSerde() throws IOException {
         }
         String json = writer.toString();
         LogicalType actual = mapper.readValue(json, LogicalType.class);
-        assertEquals(logicalType, actual);
+        assertEquals(logicalType.asSerializableString(), actual.asSerializableString());
```

Comment:
since the asSerializableString can not represent all use case, we can not compare the result using this method, maybe just keep the equals compare and  add add some specical cases